app/code/local/Magmodules/Webwinkelconnect/Helper/Data.php

```diff
@@ -100,4 +100,52 @@ public function isProductReviewInviteEnabled(): bool {
         return true;
     }
 
+    public function hasCorrectCredentials(?string $webshop_id, ?string $api_key): bool {
+        if (!trim($webshop_id) || !trim($api_key)) {
+            return false;
+        }
+        if ($webshop_id != $this->getShopId() || !hash_equals($api_key, $this->getApiKey())) {
+            return false;
+        }
+        return true;
+    }
+
+    public function syncProductReview(array $product_review): array {
+        $connection = Mage::getSingleton('core/resource')->getConnection('core_write');
+        try {
+            $connection->beginTransaction();
+
+            ($review = Mage::getModel('review/review'))
+                ->setEntityPkValue($product_review['product_review']['product_id'])
+                ->setStatusId(Mage_Review_Model_Review::STATUS_PENDING)
+                ->setTitle($product_review['product_review']['title'])
+                ->setDetail($product_review['product_review']['review'])
+                ->setEntityId($review->getEntityIdByCode(Mage_Review_Model_Review::ENTITY_PRODUCT_CODE))
+                ->setStoreId(Mage::app()->getStore()->getStoreId())
+                ->setStores([Mage::app()->getStore()->getStoreId()])
+                ->setCustomerId($this->getCustomerId($product_review['product_review']['reviewer']['email']))
+                ->setNickname($product_review['product_review']['reviewer']['name'])
+                ->save();
+
+            Mage::getModel('rating/rating')
+                ->setRatingId(Mage::getStoreConfig('webwinkelconnect/product_review_invites/rating'))
+                ->setReviewId($review->getId())
+                ->addOptionVote($product_review['product_review']['rating'], $product_review['product_review']['product_id']);
+            $review->aggregate();
+            $review->setCreatedAt(DateTime::createFromFormat('Y-m-d H:i:s', $product_review['product_review']['created']));
+            $review->save();
+
+            $connection->commit();
+            return ['message' => $review->getId(), 'code' => 200];
+        } catch (Exception $e) {
```

Comment:
In case of an exception you need to rollback the transaction that you started:

~~~php
$connection->rollBack();
~~~